### PR TITLE
#1990 added route, service, and controller method to delete unit

### PR DESCRIPTION
**Pull request by superhvarn**
## Changes

added route, service, and controller method to delete unit

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes # (issue #1990)

### CODE REVIEW

File: src/backend/src/services/projects.services.ts [hunk 1]
```diff
@@ -696,6 +696,35 @@ export default class ProjectsService {
...
+      where: {
+        name
+      }
+    });
+
+    if (!unit) {
+      throw new NotFoundException('Unit', name);
+    }
+
+    const deletedUnit = await prisma.unit.delete({
```
Comment (Aaryan1203) [inline src/backend/src/services/projects.services.ts]: no need to save this to a constant, you can just call the function

File: src/backend/tests/projects.test.ts [hunk 2]
```diff
@@ -925,6 +925,29 @@ describe('Projects', () => {
...
+
+      await expect(ProjectsService.deleteUnit(batman, 'New Unit')).rejects.toThrow(
+        new NotFoundException('Unit', 'New Unit')
+      );
+    });
+
+    test('Delete Unit works', async () => {
+      vi.spyOn(prisma.unit, 'findUnique').mockResolvedValue(toolMaterial);
+      vi.spyOn(prisma.unit, 'delete');
+      const deletedUnit = await ProjectsService.deleteUnit(superman, 'NERSoftwareTools');
```
Comment (Aaryan1203) [inline src/backend/tests/projects.test.ts]: same thing here

File: src/backend/tests/projects.test.ts [hunk 3]
```diff
@@ -925,6 +925,29 @@ describe('Projects', () => {
...
+      expect(prisma.unit.delete).toBeCalledTimes(1);
```
Comment (Aaryan1203) [inline src/backend/tests/projects.test.ts]: to ensure that it is deleted, try to find the unit and ensure that it gives you a notfoundexception

**Review by Aaryan1203 — CHANGES_REQUESTED** [review]
Left a couple of comments

**Review by Aaryan1203 — DISMISSED** [review]
LGTM!

File: src/backend/src/services/projects.services.ts [hunk 4]
```diff
@@ -696,6 +696,35 @@ export default class ProjectsService {
...
+      where: {
+        name
+      }
+    });
+
+    if (!unit) {
+      throw new NotFoundException('Unit', name);
+    }
+
+    await prisma.unit.delete({
```
Comment (RChandler234) [inline src/backend/src/services/projects.services.ts:721]: we do actually wanna save this value in a constant then return it

**Review by RChandler234 — CHANGES_REQUESTED** [review]

**Review by RChandler234 — CHANGES_REQUESTED** [review]
LGTM 🔥

**Pull request merged**